schema.graphql

```diff
@@ -107,6 +121,162 @@ type UnprocessedEntity @entity {
   block: Block
 }
 
+type Service @entity {
+  id: ID!
+  name: String!
+  computeUnitsPerRelay: BigInt!
+  owner: Account!
+  addServiceMsgs: [AddServiceMsg] @derivedFrom(field: "service")
+}
+
+type AppStakeMsg @entity {
+  id: ID!
+  stake: Coin!
+  transaction: Transaction!
+  block: Block!
+  application: Application! @index
+}
+
+type AppStakeMsgService @entity {
+  id: ID!
+  appStakeMsg: AppStakeMsg!
+  service: Service!
+}
```

Comment:
I don't understand why these are distinct types. I would've expected to see `AppStakeMsg` with both `stake` and `services` properties.

NOTE: Currently, applications can only stake for a single service despite the data structure supporting multiple.

```suggestion
type AppStakeMsg @entity {
  id: ID!
  stake: Coin!
  services: [Service] @derivedFrom(field: "id")!
  transaction: Transaction!
  block: Block!
  application: Application! @index
}
```

Reply:
Applying your suggested change makes the `yarn codegen` to throw the following error: `Please check entity Application with field services has correct relation with entity Service`.

This error happens because it expect the Service entity to have the ID of the application in the id field. If we do it then to establish a relationship between a service and many applications we would need to save several times the same service but changing the applicationId.

That's why I think we need many-to-many relationship here.